bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/persistence/PersistenceResource.java

```diff
@@ -232,10 +313,160 @@ private Response getItemHistoryBean(String serviceName, String itemName, String
         for (Map.Entry<String, PersistenceService> service : persistenceServices.entrySet()) {
             ServiceBean serviceBean = new ServiceBean();
             serviceBean.name = service.getKey();
+            PersistenceService persistence = service.getValue();
+            if (persistence instanceof ModifiablePersistenceService) {
+                serviceBean.classname = "ModifiablePersistenceService";
+            } else if (persistence instanceof QueryablePersistenceService) {
+                serviceBean.classname = "QueryablePersistenceService";
+            } else {
+                serviceBean.classname = "PersistenceService";
+            }
 
             beanList.add(serviceBean);
         }
 
         return beanList;
     }
+
+    private Response getServiceItemList(String serviceName) {
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
+            // TODO: Add handler for default service once this is available in ESH
```

Comment:
> We could add a PersistenceExtensions.getDefaultService() method

For the moment, this is probably the simplest, if the theory above did not work out...
Another option would be to introduce a PersistenceConfigService which gets configured and the other services depend on it - but feels like too much overhead...